Storage/Storage/OpenTabs.swift

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+//
+
+import Foundation
+
+public protocol OpenTabs {
+    init(files: FileAccessor)
+    
+    func persist()
+    
+    func canRestore() -> Bool
+}
+
+public class FileOpenTabs : OpenTabs {
+    private let files: FileAccessor
+
+    let restoreFile = "open-tabs.json"
```

Comment:
In the same naming vein, how about "sessionstore.json"? Making it more like Desktop and Android.